floss/language/go/extract.py

```diff
@@ -0,0 +1,431 @@
+
+            for m in build_id_regex.finditer(section_data):
+                addr = m.start() + pe.OPTIONAL_HEADER.ImageBase + section_va
+                try:
+                    string = m.group(0).decode("utf-8")
+                    yield StaticString(string=string, offset=addr, encoding=StringEncoding.ASCII)
+                except UnicodeDecodeError:
+                    continue
+                break
+
+        if section_name == ".text":
+            # Extract long strings
+            section_va = section.VirtualAddress
+            section_size = section.SizeOfRawData
+            section_data = section.get_data(section_va, section_size)
+
+            if alignment == 0x10:
+                for m in longstring64.finditer(section_data):
+                    format = "<I"
+                    s_off = struct.unpack(format, m.group(1))[0]
+                    s_size = struct.unpack(format, m.group(2))[0]
+
+                    s_rva = s_off + m.start() + section_va + 7
+                    addr = m.start() + pe.OPTIONAL_HEADER.ImageBase + section_va
+                    try:
+                        string = pe.get_string_at_rva(s_rva, s_size).decode("utf-8")
+                        if string.isprintable() and string != "" and len(string) >= min_length:
+                            yield StaticString(string=string, offset=addr, encoding=StringEncoding.ASCII)
+                    except UnicodeDecodeError:
+                        continue
+
+                for m in longstring64_2.finditer(section_data):
+                    s_off = struct.unpack("<I", m.group(3))[0]
+                    s_size = struct.unpack("<B", m.group(1))[0]
+
+                    s_rva = s_off + m.end() + section_va
+                    addr = m.start() + pe.OPTIONAL_HEADER.ImageBase + section_va
+                    try:
+                        string = pe.get_string_at_rva(s_rva, s_size).decode("utf-8")
+                        if string.isprintable() and string != "" and len(string) >= min_length:
+                            yield StaticString(string=string, offset=addr, encoding=StringEncoding.ASCII)
+                    except UnicodeDecodeError:
+                        continue
+
+                for m in longstring64_3.finditer(section_data):
+                    s_off = struct.unpack("<I", m.group(2))[0]
+                    s_size = struct.unpack("<B", m.group(1))[0]
+
+                    s_rva = s_off + m.end() + section_va + 13
+                    addr = m.start() + pe.OPTIONAL_HEADER.ImageBase + section_va
+                    try:
+                        string = pe.get_string_at_rva(s_rva, s_size).decode("utf-8")
+                        if string.isprintable() and string != "" and len(string) >= min_length:
+                            yield StaticString(string=string, offset=addr, encoding=StringEncoding.ASCII)
+                    except UnicodeDecodeError:
+                        continue
```

Comment:
there's a lot of duplication here except the changing offset, let's create a function for it

Reply:
the offsets are currently unclear to me, can you add a comment explaining them, please?